pkg/kubelet/dockertools/docker.go

```diff
@@ -534,19 +533,29 @@ var (
 	ErrContainerCannotRun = errors.New("Container cannot run")
 )
 
-func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) (*api.ContainerStatus, error) {
+// Internal information kept for containers from inspection
+type containerStatusResult struct {
+	status api.ContainerStatus
+	podIP  string
+	err    error
+}
+
+func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) *containerStatusResult {
```

Comment:
WDYT if we meet in the middle, we have containerStatusResult with a status
and an ip (no error). The function returns (containerStatusResult, error).
I think this is more natural for Go and shouldn't have too many return
types.

On Thu, Mar 19, 2015 at 9:55 AM, Dawn Chen <notifications@github.com> wrote:

> In pkg/kubelet/dockertools/docker.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5631#discussion_r26773710>
> :
>
> > @@ -534,19 +533,29 @@ var (
> >  	ErrContainerCannotRun = errors.New("Container cannot run")
> >  )
> >
> > -func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) (*api.ContainerStatus, error) {
> > +// Internal information kept for containers from inspection
> > +type containerStatusResult struct {
> > +	status api.ContainerStatus
> > +	podIP  string
> > +	err    error
> > +}
> > +
> > +func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) *containerStatusResult {
>
> I know. Another one I want to change is ParseDockerName :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5631/files#r26773710>
> .
>